docs/deprecations.md

```diff
@@ -6,6 +6,13 @@ layout: Doc
 
 # Serverless Framework Deprecations
 
+<a name="AWS_EVENT_BRIDGE_CUSTOM_RESOURCE_DEPLOYMENT"><div>&nbsp;</div></a>
+
+## AWS EventBridge lambda event triggers
+
+Starting with v3.0.0 AWS EventBridge lambda event triggers and all associated EventBridge resouurces will be deployed using native CloudFormation instead of a CustomResource that used a lambda to deploy them via the AWS SDK/API.
+
+To use this prior to deprecation set `useNativeCloudFormation: true` in lambda EventBridge trigger definition. This is a per eventt trigger config property.
```

Comment:
I agree that a provider level config should be included, I think event level config can add value too. Given that there isn't a clear, clean or any upgrade path for `EventBridge` resources already deployed with the framework allowing for a more granular adoption of CloudFormation might be wise. The underlying resources for a given stack will be replaced when switching to CloudFormation which could cause a chunk of pain with any dependencies between stacks. My concerns may be unfounded but the destruction and creation of new resources makes me wary.

Out of scope here but I have also thought that perhaps using the `CustomResource` deployment pattern should be persisted beyond the next major version bump but be opt in instead of opt out. As this is the first transition from `CustomResource` to `CloudFormation` for the framework it is worth thinking about, any future transitions may concern importable resources which would be better but that is not the case here.

Reply:
@stuartforrest-infinity do you think you will need a support for granular adoption (so deploy from same stack with both kinds of deployments) ?

I would go for event level support, only if we have a confirmed need for such support or that it's very straightforward to implement and there seems no risk of implications of deploying both kinds with same stack

Reply:
Let's go for provider level config and just be very clear in the deprecation/docs that a switch of deployment pattern will destroy existing resources.

Reply:
I fully agree 👍